test/p2p_c_version/offerer.c

```diff
@@ -0,0 +1,317 @@
+
+ #include <rtc/rtc.h>
+
+ #include <stdbool.h>
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+ #include <unistd.h> // for sleep
+ #include <ctype.h>
+ char* state_print(rtcState state);
+ char* rtcGatheringState_print(rtcState state);
+typedef struct {
+	rtcState state;
+	rtcGatheringState gatheringState;
+	int pc;
+	int dc;
+	bool connected;
+} Peer;
+
+Peer *peer = NULL;
+static void descriptionCallback(const char *sdp, const char *type, void *ptr);
+
+static void candidateCallback(const char *cand, const char *mid, void *ptr);
+
+static void stateChangeCallback(rtcState state, void *ptr);
+
+static void gatheringStateCallback(rtcGatheringState state, void *ptr);
+
+static void openCallback(void *ptr);
+
+static void closedCallback(void *ptr);
+
+static void messageCallback(const char *message, int size, void *ptr);
+
+static void deletePeer(Peer *peer);
+int all_space(const char *str);
+
+
+int main(int argc, char **argv){
+      rtcInitLogger(RTC_LOG_DEBUG);
+
+	// Create peer
+	rtcConfiguration config;
+	memset(&config, 0, sizeof(config));
+
+      Peer *peer = (Peer *)malloc(sizeof(Peer));
+	if (!peer){
+
+            printf("Error allocating memory for peer\n");
+            deletePeer(peer);
+
+      }
+	memset(peer, 0, sizeof(Peer));
+
+      printf("Peer created\n");
+
+      // Create peer connection
+      peer->pc = rtcCreatePeerConnection(&config);
+	rtcSetUserPointer(peer->pc, peer);
+      rtcSetLocalDescriptionCallback(peer->pc, descriptionCallback);
+	rtcSetLocalCandidateCallback(peer->pc, candidateCallback);
+	rtcSetStateChangeCallback(peer->pc, stateChangeCallback);
+	rtcSetGatheringStateChangeCallback(peer->pc, gatheringStateCallback);
+
+      // Since this is the offere, we will create a datachannel
+	peer->dc = rtcCreateDataChannel(peer->pc, "test");
+
+	rtcSetOpenCallback(peer->dc, openCallback);
+
+
+	rtcSetClosedCallback(peer->dc, closedCallback);
+
+	rtcSetMessageCallback(peer->dc, messageCallback);
+
+
+      sleep(1);
+
+	bool exit = false;
+
+	while (!exit) {
+
+		printf("\n");
+		printf("***************************************************************************************\n");
+
+	     // << endl
+	     printf("* 0: Exit /"
+	     		" 1: Enter remote description /"
+	     		" 2: Enter remote candidate /"
+	     		" 3: Send message /"
+	     		" 4: Print Connection Info *\n"
+	     		"[Command]: ");
+
+		int command = -1;
+            if (scanf("%d", &command)){
+
+		}else {
+			break;
+		}
+		 while ((c = getchar()) != '\n' && c != EOF) { }
```

Comment:
Same here

Reply:
it was a mistake, I am new to github, I wwill fix it and put it back again

Reply:
Sure no problem

Reply:
I have tested it and it works, I pasted all the code again, Also, I noticed that I had to set but the offerer and answer type to offer in the setlocaldescription in case 1, when I set the answerer to answer, the candiate part does not seem to work. maybe it's a conceptual knowledge I am missing or both of them have to be set to "offer" type.

Reply:
Actually, it's reversed, the answerer receives the offer, so it should set "offer" in `setRemoteDescription`, and the offerer receives the answer, so it should set "answer".

As I side note, I guess you use the website to make the modifications, which is a pain, in practice you can just push more commits on the branch and they will appear here :)

Reply:
Didnot know that, thank you very much jajajaj. And thank you for the clarification of the answer and offer, it works